### DISCUSSION TITLE
How to move scene thumbnails where are those actually stored

### DISCUSSION THREAD

**Discussion by disconnect5852** [stashapp/stash#2875] [Q&A] [answered]
Ive just trying to move my stash db from my gaming windows pc to my linux server's docker instance.
Ive copied the sqlite db file and i can see all the data correctly in new stash after a full scan (which updated the windows paths to linux paths), except the scene thumbnails. Some of them are re generated (not where i set them in the windows stash), some of them not at all.
I copied the things from windows screenshot folder to the new screenshot folder, expecting that it will prompt overwrite, due to the source video files are the same so the screenshots are saved by hash, but it isnt.
Also i see that in sqlite db there is a table (scene_thumbnails or something) with jpeg image column.
So where are these actually stored? In generated/screenshot folder, or in db?
And how can i move the scene thumbnails from windows to linux?

**Comment by disconnect5852** [comment]
Noone knows?

**Comment by MrX292** [comment]
some are in the db and some in the screenshot folder. can you try a different browser or reset the browser cache

**Reply by disconnect5852** [reply]
> some are in the db and some in the screenshot folder. can you try a different browser or reset the browser cache

It has nothing to do with browser.
The scene covers are regenerated while a new scan to fix the paths. The log wrote "generating screenshot" for every file, and reset all of the scene covers. While the cover screenshots also moved to the new stash instance.
However the newly generated screenshot file names are half as short as the windows ones. Both look like hashes, but different length hashes.

**Reply by MrX292** [reply]
can you check on windows and Linux what your settings are under System - Hashing

**Reply by disconnect5852** [reply]
I will check it, but the scenes and scenes and scenes_covers tables are exactly the same on windows and linux db, except of course the paths.
The cover pictures also the same and correct in the table, but it seems the file covers are served, which is regenerated.

Perhaps if i delete the generated screenshot files, stash will serve them from scenes_covers table?

**Comment by disconnect5852** [comment, marked as answer]
So the solution was: windows was set for the same hashing, but the covers generated beforeoshash become available, and migration from old hash names to new hash naming at files wasnt automatic. Now i clicked on Task/migrations/rename generated file solved the problem, after that i copied to linux machine, and the cover are the same.

However i still dont understand why the covers are in db and in files too.
